### PR TITLE
Introduce compatibility with neo4j by including pull size

### PR DESCRIPTION
**Pull request by jmatak**
- Introduce pull size information needed by Neo4j

### CODE REVIEW

**Comment by jmatak** [comment]
@antaljanosbenjamin Here I am creating a map and sending it to `mg_session_pull`. Should this information stay in `mgconsole` or be pushed to `mgclient`.

File: src/utils/utils.cpp [hunk 1]
```diff
@@ -805,7 +805,17 @@ QueryData ExecuteQuery(mg_session *session, const std::string &query) {
     }
   }
 
-  status = mg_session_pull(session, nullptr);
+  auto *pull_information = mg_map_make_empty(1);
```
Comment (antaljanosbenjamin) [inline src/utils/utils.cpp:808]: This will leak without destroying the map.

File: src/utils/utils.cpp [hunk 2]
```diff
@@ -805,7 +805,17 @@ QueryData ExecuteQuery(mg_session *session, const std::string &query) {
...
+  if (!pull_information) {
+    throw utils::ClientFatalException(mg_session_error(session));
+  }
+
+  // Pulling unlimited stream of information
+  mg_value *n_val = mg_value_make_integer(-1); 
+  if (!pull_information || mg_map_insert_unsafe(pull_information, "n", n_val) != 0) {
```
Comment (antaljanosbenjamin) [inline src/utils/utils.cpp:815]:
```suggestion
  if (!mg_map_insert_unsafe(pull_information, "n", n_val) != 0) {
```
It is already checked above.

File: src/utils/utils.cpp [hunk 3]
```diff
@@ -805,7 +805,17 @@ QueryData ExecuteQuery(mg_session *session, const std::string &query) {
...
   }
 
-  status = mg_session_pull(session, nullptr);
+  auto *pull_information = mg_map_make_empty(1);
+  if (!pull_information) {
+    throw utils::ClientFatalException(mg_session_error(session));
+  }
+
+  // Pulling unlimited stream of information
+  mg_value *n_val = mg_value_make_integer(-1);
```
Comment (antaljanosbenjamin) [inline src/utils/utils.cpp:814]: If the insertion is not successful, then this value has to be destroyed.

**Review by antaljanosbenjamin — COMMENTED** [review]
It is good question whether this should be part of `mgclient` of `mgconsole`. I checked the [Bolt specification](https://neo4j.com/docs/bolt/current/bolt/message/#message-pull), and `n` has to be present. That means, `mgclient` is currently not Bolt conformant. Thus I would implement this in `mgclient`, not in `mgconsole`. Does this make sense to you?

**Comment by gitbuda** [comment]
@antaljanosbenjamin I'll merge this to make it work under `mgconsole`, at some future point we can change the `mgclient`

**Pull request merged**